### PR TITLE
Port obs-mon to orion and hercules, add lua module files to create obs-mon python ve

### LINKED ISSUE #41: Port obs-monitor to orion/hercules
Add necessary system config settings to run on orion/hercules.

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
This PR includes:

- Port Obs-Monitor to orion & hercules. 
- Addition of lua module files to create an obs-mon python virtual environment.  Note that I have the ve components temporarily installed in my own noscrub space.  I know this isn't suitable for a permanent solution but thought it would be an acceptable placeholder.
- Correct use of CFP on wcoss2.  Really this time I've got it.  Honest.
- Removed file cleanup at finish.  This needs to be done as a separate job submission which runs after the plot job finishes.  I'll open a new issue to get that done.

Lastly, as describe in the discussion in #41  hercules is now having what looks like communications issues when it runs eva.  Execution hangs when trying to resolve a url request from cartopy.  That may be a system issue, or it might be a permissions issue from the ve components which I parked in my noscrub space.  Though that location works fine on orion (and wcoss2).  I'm not quite sure how much to make of that.

Close #41

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
@EdwardSafford-NOAA is the hercules issue that it is trying to download shapefiles or something else? If shapefiles, I think they either have to be in your ~/.local (while orion and hercules share /work and /work2, they do not share ~/) or an environment variable has to be set.

File: modulefiles/obs-monitor/orion.lua
Comment (CoryMartin-NOAA) [inline modulefiles/obs-monitor/orion.lua:1]: I don't see a hercules modulefile in this PR

Comment (EdwardSafford-NOAA) [reply modulefiles/obs-monitor/orion.lua:1]: It's there now.  I hadn't fully understood the orion/hercules file system(s) (and probably still don't).  I was reusing the orion mod on hercules, which actually works better than it probably should.

**Comment by EdwardSafford-NOAA** [comment]
Yes the issue is a failed attempt to download a shape file.  Your suggestion was spot on -- my `~/.local/shared/cartopy` directory on hercules did not include shape files.  I used a brute force approach and copied my `~/.local/shared` directory from orion over to hercules and that fixed the problem.  

What is the proper way of loading the `~/.local/shared` directory?  Was that populated on orion when I did a pip install of eva and supporting packages to `~/.local`?

**Comment by EdwardSafford-NOAA** [comment]
I have it figured out.  The cartopy documentation is a bit thin but eventually I learned that I can set an environment variable `CARTOPY_DATA_DIR` to the `python/envs/obs-mon/shared/cartopy` directory.  That directory must then contain the expected directory tree (in this case `shapefiles/natural_earth/physical`) and the files contained in `ne_110m_coastline.zip`.  Note the zip file must be **_expanded_**; the zip file itself is **_not_** the solution (yeah, that took a while).

I have that working on hercules and will test it on hera, orion, and wcoss2.  I'll advise when all is ready for review.

One point of curiosity, at least to me, remains.  On all platforms save hercules, `cartopy `is able to download `https://naturalearth.s3.amazonaws.com/110m_physical/ne_110m_coastline.zip`.  That effort times out on hercules.  But I was able to retrieve that file from the command line on hercules using `wget`.  So the url fetch restriction must be applied to the batch nodes but not the interactive nodes.  I understand why that might be so, but why it's inconsistent between orion and hercules is an interesting question.

**Comment by CoryMartin-NOAA** [comment]
Ah, yeah actually I'm surprised Orion lets you download on batch nodes. Hera definitely does not. I forgot about that environment variable, I actually think Rahul is the one who added that to cartopy.

**Comment by EdwardSafford-NOAA** [comment]
An update.  
1. I was wrong, the batch nodes on `hera` and `wcoss2` cannot download the needed zip file.  `Cartopy ` was apparently picking up the shape file from an earlier manual install I did an earlier stage of development on those two machines.
2. `wcoss2` is not playing nicely.  While `CARTOPY_DATA_DIR` is available to the plot script it's not getting used because reasons.  The fun continues unabated.

**Comment by CoryMartin-NOAA** [comment]
For 2, is this similar to the issues from before (I forget the specifics) of WCOSS environments being different at login vs batch (I remember PYTHONPATH being finicky)

**Comment by EdwardSafford-NOAA** [comment]
I suspect so.  Batch jobs on `wcvoss2` don't inherit any modules loaded in the submitting script's environment.  Environment variables are handled a bit differently too.  I'm dumping the value of `CARTOPY_DATA_DIR` in the python plot script and it's pointing to the right place.  It just isn't being used.  I've got a few ideas and will continue to grind away at this.

**Comment by EdwardSafford-NOAA** [comment]
Excelsior!  It was an own goal but one that was only possible on wcoss2.  On wcoss2 the environment variables set in a script that submits a batch job are not automatically included in the batch job's environment.  It's possible to include them all by using a -V option but that is frowned upon by ops because it taxes system resources.  Instead a list of necessary environment vars is to be submitted in the job submission command.  

I effed up and missed a comma in that list, but rather than throwing an error it corrupted the definition of `CARTOPY_DATA_DIR`, but in a way that was not immediately obvious when the value was dumped out.  That took longer than it should have but things like this always seem easy when you know the answer.  I'll clean all this up tomorrow morning and get it ready for re-review.

**Comment by EdwardSafford-NOAA** [comment]
All set now for re-review.

File: driver/runObsMon.sh [hunk 1]
```diff
@@ -105,8 +105,10 @@ if [[ ! -d ${logdir} ]]; then mkdir -p ${logdir}; fi
 logfile="${logdir}/OM_setup.log"
 if [[ -e ${logfile} ]]; then rm ${logfile}; fi
 
+echo MACHINE_ID: $MACHINE_ID
```
Comment (CoryMartin-NOAA) [inline driver/runObsMon.sh]: is this for debugging or should we keep it?

File: parm/OM_config [hunk 2]
```diff
@@ -9,30 +9,64 @@ export JOBSobsmon=${om_dir_root}/jobs
...
 
-      ptmp="/scratch2/NCEPDEV/stmp3/$USER"
-      stmp="/scratch2/NCEPDEV/stmp1/$USER"
+      ptmp="/scratch2/NCEPDEV/stmp3/${USER}"
+      stmp="/scratch2/NCEPDEV/stmp1/${USER}"
       queue=""
       project=""
       account="da-cpu"
       aprun_py="python"
+      module load obs-monitor/hera
```
Comment (CoryMartin-NOAA) [inline parm/OM_config]: why not a single `module load obs-monitor/${MACHINE_ID}` line?

**Review by CoryMartin-NOAA — COMMENTED** [review]
couple of minor comments, otherwise this looks good to me

File: driver/runObsMon.sh (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply driver/runObsMon.sh]: It's gone.

File: parm/OM_config (hunk 2 above)
Comment (EdwardSafford-NOAA) [reply parm/OM_config]: Done.

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks great, thanks!

**Pull request merged**